wikimedia/wikipedia-ios
# Native Editor - Add edit summary tags

**Phabricator:** 
https://phabricator.wikimedia.org/T355265 (First checkbox).

### Notes
This PR appends tags to the edit summaries of all edits in the app, to help us keep better track of the source of an edit in the iOS app via the MediaWiki edit history.

**Note:** I have gone through these test steps myself and they are a bit tedious, so we can lean on QA for more extensive testing. I'm listing out what I tested for documentation.

**Also note:** The commit in [72e6825](https://github.com/wikimedia/wikipedia-ios/pull/4738/commits/72e6825a5605672fba0b2190e419b209a032b2ce) is mostly undone in [603cec4](https://github.com/wikimedia/wikipedia-ios/pull/4738/commits/603cec4a7bed8cda52e4ec9322287041e9648963), if you're reviewing commit-by-commit.

### Test Steps
1. Make a Test Wiki edit in the new editor and confirm the correct tag shows up in the edit summary on Desktop. Test full page article, article section, article select text > edit contextual menu, and a talk page full source edit.
2. Go to a diff view via the article revision history screen. Undo and rollback an edit. Confirm correct tag shows up in edit summary on Desktop.
3. Repeat step 2 but coming from Watchlist.
4. Add a talk page new topic and new reply on a Test Wiki talk page. Confirm correct tag shows up in edit summary on Desktop.

### Article descriptions
I did test adding and changing an article description on both Test Wiki and Test Wikidata, though not without significant code changes to get around our lack of Test Wiki --> Test Wikidata connection. I would recommend leaning on QA for production testing here.
